src/software/multithreading/observer_test.cpp

```diff
@@ -46,3 +46,33 @@ TEST(Observer, receiveValue_value_not_yet_available)
     ASSERT_TRUE(result);
     EXPECT_EQ(202, *result);
 }
+
+TEST(Observer, getDataReceivedPerSecond_time_buffer_filled)
+{
+    TestObserver test_observer;
+    for (unsigned int i = 0; i < TestObserver::TIME_BUFFER_SIZE; i++)
+    {
+        test_observer.receiveValue(i);
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+    std::cout << test_observer.getDataReceivedPerSecond() << std::endl;
```

Comment:
nit: printouts should be removed before merging.

Reply:
oops